src/codegen.cpp

```diff
@@ -9568,7 +9586,7 @@ static jl_llvm_functions_t
             }
 
             mallocVisitStmt(sync_bytes, have_dbg_update);
-            if (toplevel || ctx.is_opaque_closure)
+            if (ctx.is_opaque_closure)
```

Comment:
IIUC, this is okay because we know that it must have been called via the jl_invoke in jl_toplevel_eval_flex, which sets and then resets this for the caller. Maybe it should have a comment here noting that?